### ISSUE TITLE
Internal error when minio using HTTPS and application is working in a multithreading mode

### ISSUE THREAD

**Issue by Qtexl2** [minio/minio#7501]
## Expected Behavior

I would like to upload any files with the same/diferent names in multithreaded applications by HTTPS.

## Current Behavior

Minio is started as a NAS gateway with SSL. I try to upload a file with the same name used five threads.  After that I catch the following exceptions:

```
io.minio.errors.ErrorResponseException: We encountered an internal error, please try again.
	at io.minio.MinioClient.executeReq(MinioClient.java:1190) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.execute(MinioClient.java:1056) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.executePut(MinioClient.java:1391) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.executePut(MinioClient.java:1413) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.putObject(MinioClient.java:3986) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.putObject(MinioClient.java:4064) ~[minio-6.0.4.jar!/:6.0.4]
	at io.minio.MinioClient.putObject(MinioClient.java:3952) ~[minio-6.0.4.jar!/:6.0.4]
```

```
okhttp3.internal.http2.StreamResetException: stream was reset: NO_ERROR
	at okhttp3.internal.http2.Http2Stream.checkOutNotClosed(Http2Stream.java:628) ~[okhttp-3.13.1.jar!/:?]
	at okhttp3.internal.http2.Http2Stream$FramingSink.emitFrame(Http2Stream.java:553) ~[okhttp-3.13.1.jar!/:?]
	at okhttp3.internal.http2.Http2Stream$FramingSink.write(Http2Stream.java:533) ~[okhttp-3.13.1.jar!/:?]
	at okio.ForwardingSink.write(ForwardingSink.java:35) ~[okio-1.17.3.jar!/:?]
	at okhttp3.internal.http.CallServerInterceptor$CountingSink.write(CallServerInterceptor.java:163) ~[okhttp-3.13.1.jar!/:?]
	at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:179) ~[okio-1.17.3.jar!/:?]
	at okio.RealBufferedSink.writeAll(RealBufferedSink.java:107) ~[okio-1.17.3.jar!/:?]
	at io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:74) ~[minio-6.0.4.jar!/:6.0.4]
```

```
okhttp3.internal.http2.StreamResetException: stream was reset: PROTOCOL_ERROR
	at okhttp3.internal.http2.Http2Stream.checkOutNotClosed(Http2Stream.java:628) ~[okhttp-3.13.1.jar!/:?]
	at okhttp3.internal.http2.Http2Stream$FramingSink.emitFrame(Http2Stream.java:553) ~[okhttp-3.13.1.jar!/:?]
	at okhttp3.internal.http2.Http2Stream$FramingSink.write(Http2Stream.java:533) ~[okhttp-3.13.1.jar!/:?]
	at okio.ForwardingSink.write(ForwardingSink.java:35) ~[okio-1.17.3.jar!/:?]
	at okhttp3.internal.http.CallServerInterceptor$CountingSink.write(CallServerInterceptor.java:163) ~[okhttp-3.13.1.jar!/:?]
	at okio.RealBufferedSink.emitCompleteSegments(RealBufferedSink.java:179) ~[okio-1.17.3.jar!/:?]
	at okio.RealBufferedSink.writeAll(RealBufferedSink.java:107) ~[okio-1.17.3.jar!/:?]
	at io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:74) ~[minio-6.0.4.jar!/:6.0.4]
```

## Possible Solution

Before we used minio without SSL and didn't have any problems. Minio was based on Linux in docker container, and this solution working also without problems. But we need minio in linux docker container with ssl.

## Steps to Reproduce (for bugs)

1.  Run minio as nas gateway (using SSL)
2.  Create minio client in java code, like this
```
MinioClient minioClient() throws InvalidPortException, InvalidEndpointException, IOException, InvalidKeyException, NoSuchAlgorithmException, InsufficientDataException, ErrorResponseException, NoResponseException, InvalidBucketNameException, XmlPullParserException, InternalException, RegionConflictException {
        MinioClient minioClient = new MinioClient(minioProp.getUrl(), minioProp.getAccountName(), minioProp.getAccountKey(), true);
        if (!minioClient.bucketExists(minioProp.getBucketName())) {
            minioClient.makeBucket(minioProp.getBucketName());
        }
        return minioClient;
    }
```
3.   Run this code in more than two threads
```
public String uploadChunk(byte[] bytes, String storageFileName) {
        try {
            logger.debug(ExceptionMessage.MESSAGE_MINIO_UPLOADING_BYTE_ARRAY.getText(), storageFileName, bytes.length);
            minioClient.putObject(minioProperty.getBucketName(), storageFileName,
                    new ByteArrayInputStream(bytes), APPLICATION_OCTET_STREAM_VALUE);
            String url = minioClient.presignedGetObject(minioProperty.getBucketName(), storageFileName, minioProperty.getExpiryTimeSec());
            logger.debug(ExceptionMessage.MESSAGE_MINIO_UPLOADED_BYTE_ARRAY.getText(), storageFileName);
            return url;
        } catch (MinioException | NoSuchAlgorithmException | XmlPullParserException | InvalidKeyException | IOException e) {
            logger.error(e.getMessage(), e);
            throw new StorageRuntimeException(String.format(ExceptionMessage.MINIO_UPLOAD_FILE_EXCEPTION.getText(), storageFileName), HttpStatus.BAD_REQUEST);
        }
    }
```
## Context

We want to achieve the same behavior for our on-premise solution like cloud solution ( amazon s3 or Azure blob storage
## Regression

I think it's a issue

## Your Environment

* minio version (`LATEST VERSION:2019-04-09T01:22:30Z`):
* Windows Server
![image](https://user-images.githubusercontent.com/13456325/55792973-7e473400-5aca-11e9-833b-0c35e2006564.png)
* Open JDK 11
* Minio SDK 6.0.4

**Comment by nitisht** [comment]
@Qtexl2 can you try with `minio-java` v6.0.6

**Comment by harshavardhana** [comment]
Closing this issue as fixed, we moved back to http 1.1 as there are many implementation level issues in http 2.0 support in Go.
